### PR TITLE
fix: remove hard reloads

### LINKED ISSUE #4182: Remove hard `location.reload` or hard navigate everywhere
Make sure we clear all local cache while logging out!

This excludes `IndexStatus.tsx`

### PR DESCRIPTION
**Pull request by neo773**
Removes hard reloads in the codebase

https://github.com/heyxyz/hey/assets/62795688/9ecfc7b8-77e5-49c9-9d17-130fefd2c9f6

## Related issues

Fixes #4182

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking small changes to existing functionality)

/claim #4182

### CODE REVIEW

File: apps/web/src/components/Common/Layout.tsx [hunk 1]
```diff
@@ -55,7 +57,7 @@ const Layout: FC<LayoutProps> = ({ children }) => {
     signOut();
     disconnect?.();
     if (reload) {
-      location.reload();
+      router.push('/');
```
Comment (bigint) [inline apps/web/src/components/Common/Layout.tsx:60]: We don't want to switch pages

File: apps/web/src/components/Common/Providers/LensSubscriptionsProvider.tsx [hunk 2]
```diff
@@ -84,7 +89,10 @@ const LensSubscriptionsProvider: FC = () => {
     // Using not null assertion because api returns null if revoked
     if (!authorizationRecordRevoked) {
       signOut();
-      location.reload();
+      disconnect?.();
+      setTimeout(() => {
+        router.push('/');
```
Comment (bigint) [inline apps/web/src/components/Common/Providers/LensSubscriptionsProvider.tsx:94]: Same here

File: apps/web/src/components/Shared/Login/WalletSelector.tsx [hunk 3]
```diff
@@ -130,7 +134,8 @@ const WalletSelector: FC<WalletSelectorProps> = ({
       const refreshToken = auth.data?.authenticate.refreshToken;
       signIn({ accessToken, refreshToken });
       Leafwatch.track(AUTH.SIWL);
-      location.reload();
+      onClose?.(false);
+      router.push('/');
```
Comment (bigint) [inline apps/web/src/components/Shared/Login/WalletSelector.tsx:138]: Here too

File: apps/web/src/components/Shared/Navbar/MenuItems.tsx [hunk 4]
```diff
@@ -19,12 +19,12 @@ const MenuItems: FC = () => {
   const currentProfile = useProfileStore((state) => state.currentProfile);
   const { id: sessionProfileId } = getCurrentSession();
 
-  if (Boolean(currentProfile)) {
+  if (currentProfile !== null) {
```
Comment (bigint) [inline apps/web/src/components/Shared/Navbar/MenuItems.tsx:22]: Just 'if current profile' works too

File: apps/web/src/components/Shared/Navbar/MenuItems.tsx [hunk 5]
```diff
@@ -19,12 +19,12 @@ const MenuItems: FC = () => {
...
     return <SignedUser />;
   }
 
   // If the currentSessionProfileId is a valid eth address, we can assume that address don't have a profile yet
-  if (isAddress(sessionProfileId)) {
+  if (sessionProfileId !== null && isAddress(sessionProfileId)) {
```
Comment (bigint) [inline apps/web/src/components/Shared/Navbar/MenuItems.tsx:27]: No need for null check

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by neo773** [comment]
@bigint 
Pushed a new commit

**Comment by bigint** [comment]
https://github.com/heyxyz/hey/assets/69431456/53b06b61-5502-4e55-8745-282d7bef5301

@neo773 It seems like some data are not updated, I guess something to do with apollo garbage collection or evict all cache on logout

**Comment by bigint** [comment]
Try with profile on testnet!

File: apps/web/src/components/Common/Layout.tsx [hunk 6]
```diff
@@ -48,23 +48,21 @@ const Layout: FC<LayoutProps> = ({ children }) => {
 
   const { id: sessionProfileId } = getCurrentSession();
 
-  const logout = (reload = false) => {
+  const logout = () => {
     resetPreferences();
     resetFeatureFlags();
     resetPro();
```
Comment (bigint) [inline apps/web/src/components/Common/Layout.tsx:54]: I guess we can move all these inside signout method itself so we can just call signout everywhere!

File: apps/web/src/components/Shared/Navbar/MenuItems.tsx [hunk 7]
```diff
@@ -18,6 +18,8 @@ export const NextLink = ({ children, href, ...rest }: Record<string, any>) => (
 const MenuItems: FC = () => {
   const currentProfile = useProfileStore((state) => state.currentProfile);
   const { id: sessionProfileId } = getCurrentSession();
+  console.log(currentProfile, 'currentProfile');
+  console.log(sessionProfileId, 'sessionProfileId');
```
Comment (bigint) [inline apps/web/src/components/Shared/Navbar/MenuItems.tsx:22]: Remove it

File: packages/lib/getStampFyiURL.ts [hunk 8]
```diff
@@ -7,6 +7,10 @@ import urlcat from 'urlcat';
  * @returns The cdn.stamp.fyi URL.
  */
 const getStampFyiURL = (address: string): string => {
+  // race condition fix when loging out
+  if (!address) {
+    return '';
```
Comment (bigint) [inline packages/lib/getStampFyiURL.ts:12]: you can return stamp url with null address aka `export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';`

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by neo773** [comment]
> Screen.Recording.2023-12-12.at.12.28.58.PM.mov 
> @neo773 It seems like some data are not updated, I guess something to do with apollo garbage collection or evict all cache on logout

Expected behaviour is navigating to home page every site does this,
It's impossible to account for the entire state of the App without it getting messy and would be out of scope of the original issue.

**Comment by bigint** [comment]
We mentioned it on the issue 🙇🏼 

> Make sure we clear all local cache while logging out!

**Comment by neo773** [comment]
@bigint 
Is soft reloading current URL acceptable?
Would reset the state with no edge cases.

**Comment by bigint** [comment]
WDYM soft reloading?

**Comment by bigint** [comment]
Will it reload the entire app?

**Comment by neo773** [comment]
Router refresh, so navigates to same URL

**Comment by bigint** [comment]
IMO apollo has lot of utils to clear cache or garbage collect stuffs!

**Comment by bigint** [comment]
```
const { resetStore } = useApolloClient()
```

**Comment by bigint** [comment]
is this good to review?

**Comment by neo773** [comment]
@bigint 
nah left a message on discord

**Comment by bigint** [comment]
Closing this, we can rethink this later!

**Pull request closed without merging**